### PR TITLE
week-5-tdd

### PR DESCRIPTION
**Pull request by josh8551021**
Required Test-Driven Development code
First commit: 32213a3

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/Comment.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package com.google.sps.servlets;
```
Comment (tzsebe) [inline portfolio/src/main/java/com/google/sps/servlets/Comment.java:1]: Hmm these don't seem relevant to the TDD stuff...

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 2]
```diff
@@ -14,10 +14,97 @@
...
+
+    // Mark unavailable times unavailable
+    Collection<String> meetingAttendees = request.getAttendees();
+    Collection<String> optionalAttendees = request.getOptionalAttendees();
+    for (Event event : events) {
+      Collection<String> eventAttendees = event.getAttendees();
+
+      for (String eventAttendee : eventAttendees) {
+        // Make sure all required attendees can make meeting
+        for (String meetingAttendee : meetingAttendees) {
```
Comment (tzsebe) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Have a look at the Collections docs. There's ways you can quickly look for overlaps (e.g. using `Collections.disjoint()`). Even without those, there's probable a `contains` method on collections, so you can likely do something like `meetingAttendees.contains(eventAttendee)`. It turns out they're using HashSets under the hood, so you get a performance boost there.

**Review by tzsebe — CHANGES_REQUESTED** [review]
Nice work, Josh!

If you have time, see if you can think of a much shorter implementation. The nice thing about TDD is, once you have a robust test corpus, you can refactor confidently.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 2 above)
Comment (mkmg) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: +1

**Review by mkmg — COMMENTED** [review]
Nice job adding extra tests! Let me know when you've updated the algorithm, and I'll take another look.

File: portfolio/src/main/java/com/google/sps/servlets/Comment.java (hunk 1 above)
Comment (josh8551021) [reply portfolio/src/main/java/com/google/sps/servlets/Comment.java:1]: Oh, yeah. That stuff is from a previous pull request.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 2 above)
Comment (josh8551021) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: I've changed the algorithm to utilize the `disjoint()` method.

**Review by tzsebe — APPROVED** [review]

**Review by mkmg — APPROVED** [review]
Nice work!
